snuba/writer.py

```diff
@@ -3,11 +3,11 @@
 logger = logging.getLogger('snuba.writer')
 
 
-def write_rows(connection, dataset, rows):
+def write_rows(connection, dataset, rows, types_check=False):
     table = dataset.get_schema().get_table_name()
     columns = dataset.get_schema().get_columns()
     connection.execute_robust("""
         INSERT INTO %(table)s (%(colnames)s) VALUES""" % {
         'colnames': ", ".join(col.escaped for col in columns),
         'table': table,
-    }, rows, types_check=False)
+    }, rows, types_check=types_check)
```

Comment:
`types_check` is used [here](https://github.com/getsentry/snuba/blob/72a98c998c2af9e510cc8027a899121a60ea4085/tests/base.py#L194).

This is still going to need to be removed at some point — the HTTP interface doesn't implement this. Technically, this isn't implemented as part of the native protocol either: this is implemented in [the driver](https://github.com/mymarilyn/clickhouse-driver/blob/73807b00a97870e09447b98da6a50b55710233cb/clickhouse_driver/block.py#L112-L131), and a lot less useful than I expected, as it's just making sure that the row objects are [one of the supported types](https://github.com/mymarilyn/clickhouse-driver/blob/73807b00a97870e09447b98da6a50b55710233cb/clickhouse_driver/block.py#L33-L35), not that the column values are of the correct type. If we want to maintain this behavior we need to implement it ourselves.

Personally, I think this is of questionable utility, especially if we are able to get better type checking through mypy here eventually. I think if we need type checking, we need type checking to make sure the column values that we're inserting meet the column types in the schema, not what is happening here.

Reply:
> This is still going to need to be removed at some point

I just went ahead and removed it, on second thought, based on the rationale described above. This was only being used in tests and I'm not sure it was really providing much benefit.

Reply:
There was another case where this is useful:
and it is when building the DDL for new datasets. When you try to ingest an event for a new dataset, types_check gives you some more explicit error messages telling you which column had the wrong type and why. 

Anyway, once your dataset works it is not very useful anymore.
Anyway, if the HTTP interface does not support it I guess we don't have that much of a choice.

Reply:
Oops, you're right — it looks like this [also implemented in/for `Column` classes as well](https://github.com/mymarilyn/clickhouse-driver/blob/master/clickhouse_driver/columns/base.py#L50). Not sure how I missed that last night.

Regardless, my expectation would be that the HTTP interface will still reject a write with a type mismatch (though the error might not be as useful as it could be.) I think we're going to need eventually do some of our own validation on the input data (this kind of goes back to the discussion about `group_id` being `0` or `null`) — it seems to me like that would be the right place to handle input validation instead.